Backport master into release/13

TrueBrain opened the pull request:
## Description
Backport of all closed Pull Requests labeled as 'backport requested' into `release/13`.
- https://github.com/OpenTTD/OpenTTD/pull/10879
- https://github.com/OpenTTD/OpenTTD/pull/10919
- https://github.com/OpenTTD/OpenTTD/pull/10920
- https://github.com/OpenTTD/OpenTTD/pull/10928
- https://github.com/OpenTTD/OpenTTD/pull/10926
- https://github.com/OpenTTD/OpenTTD/pull/10934
- https://github.com/OpenTTD/OpenTTD/pull/10932
- https://github.com/OpenTTD/OpenTTD/pull/10933
- All language changes
<!-- Backported: 10879,10919,10920,10928,10926,10934,10932,10933 -->

PeterN committed:
Codechange: Return fontcache font name as std::string.

PeterN committed:
Change: Include font style in font name for Freetype.

TrueBrain committed:
Fix: [SDL] unify the way X11 and Wayland handle mouse events

Basically, we drop RelativeMode completely, and use the same trick
as used by the Windows driver: read all motion events till the last
one, and use that as value.

TrueBrain committed:
Codechange: simplify UpdateCursorPositionRelative

The function is only called with fix_at=true, so don't support
the other cases.

TrueBrain committed:
Codechange: remove queue_wrap / last_position from mouse movement

No backend uses it anymore, so also no longer any need to support
it.

TrueBrain committed:
Add: [Linux] change default scroll mode to non-mouse-lock (#10920)

Wayland doesn't support mouse warping, X11 only for native
systems (so not for remote desktop, WSLg, etc), and emscripten
neither without complications. All these cannot offer a
mouse-lock.

rubidium42 commented on src/lang/english.txt:
```diff
@@ -1530,7 +1530,7 @@ STR_CONFIG_SETTING_LINKGRAPH_COLOURS_GREY_TO_RED                :Grey to red
 
 STR_CONFIG_SETTING_SCROLLMODE                                   :Viewport scroll behaviour: {STRING2}
-STR_CONFIG_SETTING_SCROLLMODE_HELPTEXT                          :Behaviour when scrolling the map
+STR_CONFIG_SETTING_SCROLLMODE_HELPTEXT                          :Behaviour when scrolling the map. The "mouse position locked" options don't work on all systems, like web-based versions, touchscreens, Linux with Wayland, and others
```
Do we maybe want to wait until the next run of the web translator so translators at least have a chance to update the string before the release?

TrueBrain replied:
We need to wait for at least 1 more eints round before we make 13.2, yes. But in general we should do a language backport just before release.
We can force an eints run early, but the question is more: when do we want to release 13.2 :)

TrueBrain committed:
Fix: disable hardware acceleration when GPU driver crashed the game last attempt (#10928)

PeterN committed:
Fix #10502: Refit engine before attaching free wagons. (#10926)

Caused by incorrect order of operations when buying a train engine with refit and attaching free wagons.

TrueBrain committed:
Fix: crash when not even a single row fits for dropdowns on low resolution screens (#10934)

TrueBrain committed:
Fix: crash when window can't be placed on low resolution screens. (#10932)

Co-authored-by: Jonathan G Rennison <j.g.rennison@gmail.com>

TrueBrain committed:
Fix: crash with tooltip on low resolution screens (#10933)

TrueBrain committed:
Fix: [Win32] position window in center of workspace of primary display (#10942)

TrueBrain committed:
Update: Backport language changes
